toxcore/DHT.c

```diff
@@ -198,6 +198,27 @@ int to_host_family(IP *ip)
     }
 }
 
+int write_dht_packet(uint8_t *target, uint16_t target_size, const uint8_t pkt_type,
+                     const uint8_t pub_key[crypto_box_PUBLICKEYBYTES],
+                     const uint8_t nonce[crypto_box_NONCEBYTES], const uint8_t *data,
+                     const uint16_t data_size) {
+  if (!target) {
+    return -1;
+  }
+
+  if (target_size - 1 - crypto_box_PUBLICKEYBYTES - crypto_box_NONCEBYTES < data_size) {
```

Comment:
That looks like it will underflow if target_size is between 0 and 56 ( `crypto_box_PUBLICKEYBYTES + crypto_box_NONCEBYTES`), is that intended?
Maybe `s/target_size/(long)target_size)/` so the crypto_box_* constants don't convert the int expression to unsigned ?